### DISCUSSION TITLE
Process requests from certain domain especially for js

### DISCUSSION THREAD

**Discussion by doguhan** [meilisearch/product#738] [Feedback & Feature Proposal]
It is not directly possible to set domain restrictions on API keys in MeiliSearch. Can you provide this

**Comment by macraig** [comment]
Hi @doguhan, would you mind expanding on your needs? 

We could potentially provide this by adding fields in the tenant token/API Key and checking, on the engine side, that [the referer HTTP header](https://developer.mozilla.org/docs/Web/HTTP/Headers/Referer) only accepts legit requests. However, as the MDN website explains, this header can be easily faked.
